[Deepin-Kernel-SIG] [linux 6.6-y] [Deepin] deepin: config: enable DRM_SMI

opsiff opened the pull request:
deepin inclusion
category: feature

To support smi graphic card in all our desktop config.

## Summary by Sourcery

Enable DRM_SMI support in the default Deepin desktop kernel configurations across all architectures

New Features:
- Add DRM_SMI driver to support SMI graphics cards

Build:
- Update Deepin desktop defconfigs for arm64, loongarch, mips, riscv, sw_64, and x86 to enable DRM_SMI

Avenger-285714 commented:
Hi @opsiff @shy129 

If possible, I recommend enabling this configuration for MIPS and s390x as well.

For MIPS, many machines on that architecture have not yet been fully retired from the 信创 market, so enabling it would be useful.

By the way, since we already cross-compile for multiple architectures, it would also be wise to add a mips64el build check to our CI — that helps catch and fix issues early when large code changes are made.

For s390x, enabling the config lets us verify that the driver can compile on a big-endian target, which is especially important for stricter code-quality review.

Avenger-285714 commented:
/approve

opsiff committed:
deepin: config: enable DRM_SMI

deepin inclusion
category: feature

To support smi graphic card in all our desktop config.

Signed-off-by: Wentao Guan <guanwentao@uniontech.com>

opsiff committed:
deepin: drm: smidrm: ddk770: fix typo in cursor.h

deepin inclusion
category: bugfix

similar to ddk768_cursor.h and etc.
modified k to K to fix it.

Log:
2025-11-19T06:44:10.7451497Z In file included from drivers/gpu/drm/smidrm/hw770.c:11:
2025-11-19T06:44:10.7456125Z drivers/gpu/drm/smidrm/ddk770/ddk770_cursor.h:12:9: error: '_DDk770_CURSOR_H_' is used as a header guard here, followed by #define of a different macro [-Werror,-Wheader-guard]
2025-11-19T06:44:10.7469716Z    12 | #ifndef _DDk770_CURSOR_H_
2025-11-19T06:44:10.7470443Z       |         ^~~~~~~~~~~~~~~~~
2025-11-19T06:44:10.7472019Z drivers/gpu/drm/smidrm/ddk770/ddk770_cursor.h:13:9: note: '_DDK770_CURSOR_H_' is defined here; did you mean '_DDk770_CURSOR_H_'?
2025-11-19T06:44:10.7473211Z    13 | #define _DDK770_CURSOR_H_
2025-11-19T06:44:10.7473748Z       |         ^~~~~~~~~~~~~~~~~
2025-11-19T06:44:10.7474277Z       |         _DDk770_CURSOR_H_

Signed-off-by: Wentao Guan <guanwentao@uniontech.com>

opsiff committed:
deepin: drm: smidrm: fix dependency in Kconfig

deepin inclusion
category: bugfix

Log:
ERROR: modpost: "snd_ctl_add" [drivers/gpu/drm/smidrm/smifb.ko] undefined!
ERROR: modpost: "snd_pcm_new" [drivers/gpu/drm/smidrm/smifb.ko] undefined!
ERROR: modpost: "snd_card_register" [drivers/gpu/drm/smidrm/smifb.ko] undefined!
ERROR: modpost: "snd_card_free" [drivers/gpu/drm/smidrm/smifb.ko] undefined!
ERROR: modpost: "snd_pcm_lib_preallocate_pages_for_all" [drivers/gpu/drm/smidrm/smifb.ko] undefined!
ERROR: modpost: "snd_ctl_new1" [drivers/gpu/drm/smidrm/smifb.ko] undefined!
ERROR: modpost: "snd_card_new" [drivers/gpu/drm/smidrm/smifb.ko] undefined!
ERROR: modpost: "snd_pcm_lib_malloc_pages" [drivers/gpu/drm/smidrm/smifb.ko] undefined!
ERROR: modpost: "snd_pcm_lib_ioctl" [drivers/gpu/drm/smidrm/smifb.ko] undefined!
ERROR: modpost: "snd_pcm_lib_free_pages" [drivers/gpu/drm/smidrm/smifb.ko] undefined!
WARNING: modpost: suppressed 3 unresolved symbol warnings because there were too many)

Signed-off-by: Wentao Guan <guanwentao@uniontech.com>

opsiff committed:
drm: smidrm: ddk750: fix strict-prototypes warning

deepin inclusion
category: other

Just makes clang build happy, no functional change.

Log:
2025-11-19T11:09:53.6529129Z drivers/gpu/drm/smidrm/ddk750/ddk750_2d.c:57:13: error: a function declaration without a prototype is deprecated in all versions of C [-Werror,-Wstrict-prototypes]
2025-11-19T11:09:53.6530540Z    57 | void deReset()
2025-11-19T11:09:53.6530958Z       |             ^
2025-11-19T11:09:53.6531401Z       |              void
2025-11-19T10:09:43.3657557Z drivers/gpu/drm/smidrm/ddk750/ddk750_power.c:72:27: error: a function declaration without a prototype is deprecated in all versions of C [-Werror,-Wstrict-prototypes]
2025-11-19T10:09:43.3673893Z    72 | unsigned long getPowerMode()
2025-11-19T10:09:43.3674636Z       |                           ^
2025-11-19T10:09:43.3675615Z       |                            void
2025-11-19T10:09:43.3734385Z 1 error generated.
2025-11-19T09:16:36.1737153Z drivers/gpu/drm/smidrm/ddk750/ddk750_chip.c:64:39: error: a function declaration without a prototype is deprecated in all versions of C [-Werror,-Wstrict-prototypes]
2025-11-19T09:16:36.1743256Z    64 | logical_chip_type_t ddk750_getChipType()
2025-11-19T09:16:36.1745432Z       |                                       ^
2025-11-19T09:16:36.1747272Z       |                                        void
2025-11-19T09:16:36.1752477Z drivers/gpu/drm/smidrm/ddk750/ddk750_chip.c:73:35: error: a function declaration without a prototype is deprecated in all versions of C [-Werror,-Wstrict-prototypes]
2025-11-19T09:16:36.1757687Z    73 | void ddk750_resetFrameBufferMemory()
2025-11-19T09:16:36.1760926Z       |                                   ^
2025-11-19T09:16:36.1764285Z   CC [M]  drivers/gpu/drm/mgag200/mgag200_g200se.o
2025-11-19T09:16:36.1769831Z       |                                    void
2025-11-19T09:16:36.1803138Z drivers/gpu/drm/smidrm/ddk750/ddk750_chip.c:205:21: error: a function declaration without a prototype is deprecated in all versions of C [-Werror,-Wstrict-prototypes]
2025-11-19T09:16:36.1808661Z   205 | long ddk750_initChip()
2025-11-19T09:16:36.1810285Z       |                     ^
2025-11-19T09:16:36.1811231Z       |                      void
2025-11-19T09:16:36.1811780Z 3 errors generated.
2025-11-19T07:49:35.4025494Z drivers/gpu/drm/smidrm/ddk750/ddk750_clock.c:255:27: error: a function declaration without a prototype is deprecated in all versions of C [-Werror,-Wstrict-prototypes]
2025-11-19T07:49:35.4027365Z   255 | unsigned long getChipClock()
2025-11-19T07:49:35.4028223Z       |                           ^
2025-11-19T07:49:35.4029015Z       |                            void
2025-11-19T07:49:35.4043181Z drivers/gpu/drm/smidrm/ddk750/ddk750_clock.c:379:38: error: a function declaration without a prototype is deprecated in all versions of C [-Werror,-Wstrict-prototypes]
2025-11-19T07:49:35.4073953Z   379 | unsigned long getPrimaryDispCtrlClock()
2025-11-19T07:49:35.4075443Z       |                                      ^
2025-11-19T07:49:35.4076477Z       |                                       void
2025-11-19T07:49:35.4079714Z drivers/gpu/drm/smidrm/ddk750/ddk750_clock.c:391:40: error: a function declaration without a prototype is deprecated in all versions of C [-Werror,-Wstrict-prototypes]
2025-11-19T07:49:35.4081500Z   391 | unsigned long getSecondaryDispCtrlClock()
2025-11-19T07:49:35.4082170Z       |                                        ^
2025-11-19T07:49:35.4082742Z       |                                         void
2025-11-19T07:49:35.4084394Z drivers/gpu/drm/smidrm/ddk750/ddk750_clock.c:403:29: error: a function declaration without a prototype is deprecated in all versions of C [-Werror,-Wstrict-prototypes]
2025-11-19T07:49:35.4086711Z   403 | unsigned long getMasterClock()
2025-11-19T07:49:35.4087261Z       |                             ^
2025-11-19T07:49:35.4087781Z       |                              void
2025-11-19T07:49:35.4089267Z drivers/gpu/drm/smidrm/ddk750/ddk750_clock.c:441:29: error: a function declaration without a prototype is deprecated in all versions of C [-Werror,-Wstrict-prototypes]
2025-11-19T07:49:35.4090955Z   441 | unsigned long getMemoryClock()
2025-11-19T07:49:35.4091520Z       |                             ^
2025-11-19T07:49:35.4092057Z       |                              void
2025-11-19T07:49:35.4093569Z drivers/gpu/drm/smidrm/ddk750/ddk750_clock.c:479:37: error: a function declaration without a prototype is deprecated in all versions of C [-Werror,-Wstrict-prototypes]
2025-11-19T07:49:35.4095118Z   479 | unsigned char *getMasterClockDivider()
2025-11-19T07:49:35.4096134Z       |                                     ^
2025-11-19T07:49:35.4096784Z       |                                      void
2025-11-19T07:49:35.4098342Z drivers/gpu/drm/smidrm/ddk750/ddk750_clock.c:487:41: error: a function declaration without a prototype is deprecated in all versions of C [-Werror,-Wstrict-prototypes]
2025-11-19T07:49:35.4099857Z   487 | unsigned long getTotalMasterClockDivider()
2025-11-19T07:49:35.4100531Z       |                                         ^
2025-11-19T07:49:35.4101096Z       |                                          void
2025-11-19T07:49:35.4102632Z drivers/gpu/drm/smidrm/ddk750/ddk750_clock.c:498:37: error: a function declaration without a prototype is deprecated in all versions of C [-Werror,-Wstrict-prototypes]
2025-11-19T07:49:35.4104141Z   498 | unsigned char *getMemoryClockDivider()
2025-11-19T07:49:35.4104741Z       |                                     ^
2025-11-19T07:49:35.4105297Z       |                                      void
2025-11-19T07:49:35.4106877Z drivers/gpu/drm/smidrm/ddk750/ddk750_clock.c:506:41: error: a function declaration without a prototype is deprecated in all versions of C [-Werror,-Wstrict-prototypes]
2025-11-19T07:49:35.4108577Z   506 | unsigned long getTotalMemoryClockDivider()
2025-11-19T07:49:35.4109191Z       |                                         ^
2025-11-19T07:49:35.4109784Z       |                                          void

Signed-off-by: Wentao Guan <guanwentao@uniontech.com>

opsiff committed:
drm: smidrm: fix build with clang

deepin inclusion
category: bugfix

Log:

2025-11-20T02:25:18.6651568Z   CC [M]  drivers/gpu/drm/i915/i915_pmu.o
2025-11-20T02:25:18.6810866Z drivers/gpu/drm/smidrm/ddk768/uda1345.c:142:5: error: variable 'addr' is used uninitialized whenever switch default is taken [-Werror,-Wsometimes-uninitialized]
2025-11-20T02:25:18.6811696Z   142 |     default:
2025-11-20T02:25:18.6811960Z       |     ^~~~~~~
2025-11-20T02:25:18.6812468Z drivers/gpu/drm/smidrm/ddk768/uda1345.c:147:20: note: uninitialized use occurs here
2025-11-20T02:25:18.6813024Z   147 |                    addr, &data, 1);
2025-11-20T02:25:18.6813378Z       |                    ^~~~
2025-11-20T02:25:18.6814001Z drivers/gpu/drm/smidrm/ddk768/uda1345.c:121:12: note: initialize the variable 'addr' to silence this warning
2025-11-20T02:25:18.6814576Z   121 |     u8 addr;
2025-11-20T02:25:18.6814827Z       |            ^
2025-11-20T02:25:18.6815075Z       |             = '\0'
Signed-off-by: Wentao Guan <guanwentao@uniontech.com>

Avenger-285714 commented:
As far as I know, your build servers have LLVM available, and local checks are often faster than CI.

**CI should be the "last line of defense".**

**Running checks locally also helps reduce "mailing-list spam."**

Thanks.

opsiff committed:
deepin: drm: smidrm: ddk770: fix typo in debug.h

deepin inclusion
category: bugfix

similar to ddk768_cursor.h and etc.
modified k to K to fix it.

Log:
2025-11-20T03:34:29.9620208Z In file included from drivers/gpu/drm/smidrm/ddk770/ddk770_helper.c:13:
2025-11-20T03:34:29.9622741Z drivers/gpu/drm/smidrm/ddk770/ddk770_ddkdebug.h:12:9: error: '_DDk770_DDKDEBUG_H_' is used as a header guard here, followed by #define of a different macro [-Werror,-Wheader-guard]
2025-11-20T03:34:29.9624475Z    12 | #ifndef _DDk770_DDKDEBUG_H_
2025-11-20T03:34:29.9625208Z       |         ^~~~~~~~~~~~~~~~~~~
2025-11-20T03:34:29.9626716Z drivers/gpu/drm/smidrm/ddk770/ddk770_ddkdebug.h:13:9: note: '_DDK770_DDKDEBUG_H_' is defined here; did you mean '_DDk770_DDKDEBUG_H_'?
2025-11-20T03:34:29.9647909Z    13 | #define _DDK770_DDKDEBUG_H_
2025-11-20T03:34:29.9648757Z       |         ^~~~~~~~~~~~~~~~~~~
2025-11-20T03:34:29.9649466Z       |         _DDk770_DDKDEBUG_H_
2025-11-20T03:34:29.9650159Z 1 error generated.

Signed-off-by: Wentao Guan <guanwentao@uniontech.com>

opsiff committed:
drm: smidrm: ddk770: fix build with clang

Log:
2025-11-20T04:13:03.9327710Z drivers/gpu/drm/smidrm/ddk770/ddk770_clock.c:513:62: error: converting the enum constant to a boolean [-Werror,-Wint-in-bool-context]
2025-11-20T04:13:03.9337226Z   513 |     else if( DDR4_2P4K_FD == type  ||  DDR4_2P4K_HD == type  ||  LPDDR4_2P4K_FD  ||  LPDDR4_2P4K_HD){
2025-11-20T04:13:03.9339580Z       |                                                              ^
2025-11-20T04:13:03.9342877Z drivers/gpu/drm/smidrm/ddk770/ddk770_clock.c:513:82: error: converting the enum constant to a boolean [-Werror,-Wint-in-bool-context]
2025-11-20T04:13:03.9352082Z   513 |     else if( DDR4_2P4K_FD == type  ||  DDR4_2P4K_HD == type  ||  LPDDR4_2P4K_FD  ||  LPDDR4_2P4K_HD){
2025-11-20T04:13:03.9361374Z       |                                                                                  ^
2025-11-20T04:13:03.9378153Z 2 errors generated.

Signed-Off-by: Teddy Wang <teddy.wang@siliconmotion.com>
Signed-off-by: Wentao Guan <guanwentao@uniontech.com>

opsiff committed:
deepin: drm: smidrm: fix uninitialized in ddk770_videoGetBufferSize

deepin inclusion
category: bugfix

Signed-off-by: Wentao Guan <guanwentao@uniontech.com>

opsiff committed:
deepin: drm: smidrm: fix build with clang

deepin inclusion
category: bugfix

Log:

drivers/gpu/drm/smidrm/ddk770/ddk770_hdmi.c:678:15: error: variable 'disp_index' is used uninitialized whenever 'if' condition is false [-Werror,-Wsometimes-uninitialized]
    }else if (index == INDEX_HDMI2)
              ^~~~~~~~~~~~~~~~~~~~
drivers/gpu/drm/smidrm/ddk770/ddk770_hdmi.c:683:11: note: uninitialized use occurs here
        setDCMUX(disp_index, dc);
                 ^~~~~~~~~~
drivers/gpu/drm/smidrm/ddk770/ddk770_hdmi.c:678:11: note: remove the 'if' if its condition is always true
    }else if (index == INDEX_HDMI2)
          ^~~~~~~~~~~~~~~~~~~~~~~~~
drivers/gpu/drm/smidrm/ddk770/ddk770_hdmi.c:671:5: note: variable 'disp_index' is declared here
    disp_output_t disp_index;
    ^
drivers/gpu/drm/smidrm/ddk770/ddk770_hdmi.c:695:15: error: variable 'disp_index' is used uninitialized whenever 'if' condition is false [-Werror,-Wsometimes-uninitialized]
    }else if (index == INDEX_HDMI2)
              ^~~~~~~~~~~~~~~~~~~~
drivers/gpu/drm/smidrm/ddk770/ddk770_hdmi.c:700:13: note: uninitialized use occurs here
        ClearDCMUX(disp_index);
                   ^~~~~~~~~~
drivers/gpu/drm/smidrm/ddk770/ddk770_hdmi.c:695:11: note: remove the 'if' if its condition is always true
    }else if (index == INDEX_HDMI2)
          ^~~~~~~~~~~~~~~~~~~~~~~~~
drivers/gpu/drm/smidrm/ddk770/ddk770_hdmi.c:688:5: note: variable 'disp_index' is declared here
    disp_output_t disp_index;
    ^
drivers/gpu/drm/smidrm/ddk770/ddk770_hdmi.c:712:15: error: variable 'disp_index' is used uninitialized whenever 'if' condition is false [-Werror,-Wsometimes-uninitialized]
    }else if (index == INDEX_HDMI2)
              ^~~~~~~~~~~~~~~~~~~~
drivers/gpu/drm/smidrm/ddk770/ddk770_hdmi.c:717:18: note: uninitialized use occurs here
        return GetDCMUX(disp_index);
                        ^~~~~~~~~~
drivers/gpu/drm/smidrm/ddk770/ddk770_hdmi.c:712:11: note: remove the 'if' if its condition is always true
    }else if (index == INDEX_HDMI2)
          ^~~~~~~~~~~~~~~~~~~~~~~~~
drivers/gpu/drm/smidrm/ddk770/ddk770_hdmi.c:705:5: note: variable 'disp_index' is declared here
    disp_output_t disp_index;
    ^

drivers/gpu/drm/smidrm/ddk770/ddk770_dp.c:949:2: error: misleading indentation; statement is not part of the previous 'if' [-Werror,-Wmisleading-indentation]
        if (power_on) {
        ^
drivers/gpu/drm/smidrm/ddk770/ddk770_dp.c:942:5: note: previous statement is here
    if (err < 0)
    ^
drivers/gpu/drm/smidrm/ddk770/ddk770_dp.c:1185:15: error: variable 'disp_index' is used uninitialized whenever 'if' condition is false [-Werror,-Wsometimes-uninitialized]
    }else if (index == INDEX_DP1){
              ^~~~~~~~~~~~~~~~~~
drivers/gpu/drm/smidrm/ddk770/ddk770_dp.c:1189:14: note: uninitialized use occurs here
    setDCMUX(disp_index, dc);
             ^~~~~~~~~~
drivers/gpu/drm/smidrm/ddk770/ddk770_dp.c:1185:11: note: remove the 'if' if its condition is always true
    }else if (index == INDEX_DP1){
          ^~~~~~~~~~~~~~~~~~~~~~~
drivers/gpu/drm/smidrm/ddk770/ddk770_dp.c:1180:5: note: variable 'disp_index' is declared here
    disp_output_t disp_index;
    ^
drivers/gpu/drm/smidrm/ddk770/ddk770_dp.c:1199:15: error: variable 'disp_index' is used uninitialized whenever 'if' condition is false [-Werror,-Wsometimes-uninitialized]
    }else if (index == INDEX_DP1){
              ^~~~~~~~~~~~~~~~~~
drivers/gpu/drm/smidrm/ddk770/ddk770_dp.c:1203:16: note: uninitialized use occurs here
    ClearDCMUX(disp_index);
               ^~~~~~~~~~
drivers/gpu/drm/smidrm/ddk770/ddk770_dp.c:1199:11: note: remove the 'if' if its condition is always true
    }else if (index == INDEX_DP1){
          ^~~~~~~~~~~~~~~~~~~~~~~
drivers/gpu/drm/smidrm/ddk770/ddk770_dp.c:1194:5: note: variable 'disp_index' is declared here
    disp_output_t disp_index;
    ^
drivers/gpu/drm/smidrm/ddk770/ddk770_dp.c:1213:15: error: variable 'disp_index' is used uninitialized whenever 'if' condition is false [-Werror,-Wsometimes-uninitialized]
    }else if (index == INDEX_DP1){
              ^~~~~~~~~~~~~~~~~~
drivers/gpu/drm/smidrm/ddk770/ddk770_dp.c:1217:21: note: uninitialized use occurs here
    return GetDCMUX(disp_index);
                    ^~~~~~~~~~
drivers/gpu/drm/smidrm/ddk770/ddk770_dp.c:1213:11: note: remove the 'if' if its condition is always true
    }else if (index == INDEX_DP1){
          ^~~~~~~~~~~~~~~~~~~~~~~
drivers/gpu/drm/smidrm/ddk770/ddk770_dp.c:1208:5: note: variable 'disp_index' is declared here
    disp_output_t disp_index;
    ^
